src/aind_behavior_curriculum/tmp.py

```diff
@@ -0,0 +1,80 @@
+"""
+Work in Progress Curriculum
+"""
+
+from collections import defaultdict
+
+class Task:
+    """
+    Defined in exisiting PR
+    """
+    pass
+
+class Metric:
+    """
+    Initalizes a connection to relevant database (referencing some configuration file)
+    User extends this with their own metric implementation based on data inside the database.
+    """
+    pass
+
+
+class Evaluation:
+    """
+    Serializable Callable, takes in a list of metrics that is serialized alongside.
+    """
+    pass
+
+
+# TODO: De/serialization
+class Curriculum:
+    def __init__(self) -> None:
+        TaskId = int
+        # ID tasks on Task.name, a required field.
+        # Tasks are allowed to change parameters.
+        self.tasks: dict[str, TaskId] = {}
```

Comment:
A curriculum is not a collection of tasks, it's a collection of stages. In theory you could have the same task twice. Mostly just a nomenclature clarification.

Reply:
Where did we land on whether we need a schema for a stage?

Reply:
I think I am almost convinced depending on how we deal with the modifiable attritube tag and assuming we followed the proposal that a stage is a stage of a task iif you can validate the stage instance against the task definition.